warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -1204,13 +1201,15 @@ private static void getERNodes(JexlNode node, List<ASTERNode> erNodes) {
         return nodes;
     }
     
-    protected static boolean isDelayedPredicate(JexlNode currNode) {
-        if (ASTDelayedPredicate.instanceOf(currNode) || ExceededOrThresholdMarkerJexlNode.instanceOf(currNode)
-                        || ExceededValueThresholdMarkerJexlNode.instanceOf(currNode) || ExceededTermThresholdMarkerJexlNode.instanceOf(currNode)
-                        || IndexHoleMarkerJexlNode.instanceOf(currNode) || ASTEvaluationOnly.instanceOf(currNode))
-            return true;
-        else
-            return false;
+    /**
+     * Return whether to not the provided node is an instance of a delayed predicate.
+     * 
+     * @param node
+     *            the node
+     * @return true if the node is an instance of a delayed predicate, or false otherwise
+     */
+    public static boolean isDelayedPredicate(JexlNode node) {
+        return QueryPropertyMarkerVisitor.instanceOfAny(node);
```

Comment:
I don't believe this should be changed.  isDelayedPredictate is specific to a certain set of query property markers.  BoundedRange is coming along in #926 which is specifically not a delayed predicate.  Please revert this to it original state.

Reply:
Fixed.

Reply:
Wait wait wait!  I still think we should avoid multiple recursions of the entire jexl tree.  Can we revert the revert, but update the isDelayedPredicate call to be...

```
return QueryPropertyMarkerVisitor.instanceOf(currNode, Arrays.asList(ASTDelayedPredicate.class, ExceededOrThresholdMarkerJexlNode.class,
                        ExceededValueThresholdMarkerJexlNode.class, ExceededTermThresholdMarkerJexlNode.class, IndexHoleMarkerJexlNode.class,
                        ASTEvaluationOnly.class), null);
```

I think this satisfies both my and Ivan's issue, right?

Reply:
Third time's the charm? Fixed.